Signal/src/ViewControllers/NewGroupViewController.h

```diff
@@ -2,27 +2,17 @@
 //  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
 //
 
-#import <UIKit/UIKit.h>
-#import "TSGroupModel.h"
-#import "TSGroupThread.h"
+#import "OWSConversationSettingsViewDelegate.h"
 
-@interface NewGroupViewController : UIViewController <UITableViewDelegate,
-                                                      UITabBarDelegate,
-                                                      UIImagePickerControllerDelegate,
-                                                      UINavigationControllerDelegate,
-                                                      UITextFieldDelegate>
+NS_ASSUME_NONNULL_BEGIN
 
-- (void)configWithThread:(TSGroupThread *)thread;
+@interface NewGroupViewController : UIViewController
 
-@property (nonatomic) IBOutlet UITableView *tableView;
-@property (nonatomic) IBOutlet UITextField *nameGroupTextField;
-@property (nonatomic) IBOutlet UIButton *groupImageButton;
-@property (nonatomic) IBOutlet UIView *tapToDismissView;
-@property (nonatomic) IBOutlet UILabel *addPeopleLabel;
-@property (nonatomic) UIImage *groupImage;
-@property (nonatomic) TSGroupModel *groupModel;
+@property (nonatomic, weak) id<OWSConversationSettingsViewDelegate> delegate;
 
 @property (nonatomic) BOOL shouldEditGroupNameOnAppear;
 @property (nonatomic) BOOL shouldEditAvatarOnAppear;
```

Comment:
I've separated the "new group" and "update group" out into separate classes - the common logic resides in the group view helper.